src/plot.jl

```diff
@@ -1,79 +1,83 @@
-using RecipesBase
-using StatPlots
-#################### Posterior Plot Recipies ####################
-@recipe function f(c::MCMCChain.AbstractChains,
-                   ptypes::Union{Symbol, AbstractArray{Symbol}};
-                   maxlag = round(Int, 10 * log10(length(c.range))),
-                   barbounds = (0, Inf))
-  ptypes = ptypes isa Symbol ? [ptypes] : ptypes
+export MeanPlot, AutocorPlot, HistogramPlot, DensityPlot, MixedDensityPlot, TracePlot
 
+@userplot MeanPlot
+@userplot AutocorPlot
+@userplot HistogramPlot
+@userplot DensityPlot
+@userplot MixedDensityPlot
+@userplot TracePlot
+
+@recipe function f(p::MeanPlot)
+  c, i = p.args
+  seriestype := :line
+  xaxis := "Iteration"
+  yaxis := "Mean"
+  MCMCChain.cummean(c.value[:, i, :])
+end
+
+@recipe function f(p::AutocorPlot; maxlag = nothing)
+  c, i = p.args
+  lags = 0:(maxlag === nothing ? round(Int, 10 * log10(length(c.range))) : maxlag)
+  ac = MCMCChain.autocor(c, lags=collect(lags))
+  xaxis := "Lag"
+  yaxis := "Autocorrelation"
+  ac.value[i, :, :]
+end
+
+@recipe function f(p::HistogramPlot)
+  c, i = p.args
+  seriestype := :histogram
+  xaxis := "Sample value"
+  yaxis := "Frequency"
+  c.value[:, i, :]
+end
+
+@recipe function f(p::DensityPlot)
+  c, i = p.args
+  seriestype := :density
+  xaxis := "Sample value"
+  yaxis := "Density"
+  c.value[:, i, :]
+end
+
+@recipe function f(p::TracePlot)
+  c, i = p.args
+  seriestype := :line
+  xaxis := "Iteration"
+  yaxis := "Sample value"
+  c.value[:, i, :]
+end
+
+@recipe function f(p::MixedDensityPlot, barbounds = (0, Inf))
+  c, i = p.args
+  discrete = MCMCChain.indiscretesupport(c, barbounds)
+  seriestype := discrete[i] ? :histogram : :density
+  xaxis := "Sample value"
+  yaxis := discrete[i] ? "Frequency" : "Density"
+  c.value[:, i, :]
+end
+
+@recipe function f(c::MCMCChain.AbstractChains;
```

Comment:
Perhaps tweak the style a bit (e.g. reduce title font size, axis font size)? The [one](https://github.com/TuringLang/MCMCChain.jl/issues/15#issuecomment-420339484) from PyMC3 looks quite nice actually.

Reply:
Unfortunately, we cannot set font types and sizes without depending on Plots. However, there is a nice package called PlotThemes which automatically sets several parameters according to predefined themes. I'll refer to this package if the user wants to improve the look of the plots.

Reply:
Perhaps this can be solved using `Requires`, see 

https://github.com/MikeInnes/Requires.jl